lens-cube/src/main/java/org/apache/lens/cube/parse/UnionCandidate.java

```diff
@@ -147,8 +148,100 @@ private String getToString() {
       builder.append(candidate.toString());
       builder.append(", ");
     }
-    builder.delete(builder.length()-2, builder.length());
+    builder.delete(builder.length() - 2, builder.length());
     builder.append("]");
     return builder.toString();
   }
+
+  private Map<Candidate, TimeRange> getTimeRangeForChildren(TimeRange timeRange) {
+    Collections.sort(childCandidates, new Comparator<Candidate>() {
+      @Override
+      public int compare(Candidate o1, Candidate o2) {
+        return o1.getCost() < o2.getCost() ? -1 : o1.getCost() == o2.getCost() ? 0 : 1;
+      }
+    });
+
+    Map<Candidate, TimeRange> candidateTimeRangeMap = new HashMap<>();
+    // Sorted list based on the weights.
+    Set<TimeRange> ranges = new HashSet<>();
+
+    ranges.add(timeRange);
+    for (Candidate c : childCandidates) {
+      TimeRange.TimeRangeBuilder builder = getClonedBuiler(timeRange);
+      TimeRange tr = resolveTimeRange(c, ranges, builder);
+      if (tr != null) {
```

Comment:
Can union children can not return null for time range ?  Aren't all children expected to participate ?